connect/runtime/src/main/java/org/apache/kafka/connect/storage/FileOffsetBackingStore.java

```diff
@@ -78,6 +85,27 @@ private void load() {
                 ByteBuffer key = (mapEntry.getKey() != null) ? ByteBuffer.wrap(mapEntry.getKey()) : null;
                 ByteBuffer value = (mapEntry.getValue() != null) ? ByteBuffer.wrap(mapEntry.getValue()) : null;
                 data.put(key, value);
+                if (key != null) {
+                    try {
+                        // The topic parameter is irrelevant for the JsonConverter which is the internal converter used by
+                        // Connect workers.
+                        List<Object> keyValue = (List<Object>) keyConverter.toConnectData("", key.array()).value();
+                        // The key should always be of the form [connectorName, partition] where connectorName is a
+                        // string value and partition is a Map<String, Object>
+                        String connectorName = (String) keyValue.get(0);
+                        Map<String, Object> partition = (Map<String, Object>) keyValue.get(1);
+                        if (!connectorPartitions.containsKey(connectorName)) {
+                            connectorPartitions.put(connectorName, new HashSet<>());
+                        }
+                        if (value == null) {
+                            connectorPartitions.get(connectorName).remove(partition);
```

Comment:
Do we ever remove keys from the `connectorPartitions` itself? It seems like this leads to an infinitely-growing map.

I feel like we've had a similar discussion on other PRs and the conclusion has been "Yes but practically speaking it's not an issue"; if so (and if this scenario can be treated the same way), feel free to just drop a link and resolve this comment.